### PR TITLE
client-go: Use log levels that match logging conventions

### LINKED ISSUE #87867: client-go: Use log levels that match logging conventions
/kind cleanup
/sig instrumentation

**What happened**:
SIG Instrumentation has published logging conventions for use in Kubernetes.
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md
The existing code has calls in the range 0 to 10 and still remains to be normalized to match the conventions.

**What you expected to happen**:
All klog.V() calls use values in the 0-5 range. So when a user specifies a verbosity of 5 they will receive the highest level of detail.
All uses of klog.V() (not just those over 5) should be reviewed for alignment with the logging conventions and adjusted as makes sense.

List of files to review (instances where log level > 5):
vendor/k8s.io/client-go/discovery/cached/disk/cached_discovery.go
vendor/k8s.io/client-go/rest/request.go
vendor/k8s.io/client-go/tools/clientcmd/loader.go
vendor/k8s.io/client-go/tools/remotecommand/v1.go
vendor/k8s.io/client-go/transport/round_trippers.go

**How to reproduce it (as minimally and precisely as possible)**:
Run with a log level of 5, and again with a log level of 10 and compare the resulting logs.
Note that many messages above 5 may only appear under certain error conditions.

**Anything else we need to know?**:
This is not a critical bug and most users will not notice any change.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.17 and older
- Cloud provider or hardware configuration: N/A
- OS (e.g: `cat /etc/os-release`): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Network plugin and version (if this is a network-related bug): N/A
- Others:

### PR DESCRIPTION
**Pull request by ryanrod**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind cleanup

**What this PR does / why we need it**:
This PR examines all uses of klog.V() and adjusts them according to the recommended values located [here.](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md)

**Which issue(s) this PR fixes**:

Fixes #87867

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/rest/request.go [hunk 1]
```diff
@@ -1007,11 +1007,11 @@ func (r *Request) transformResponse(resp *http.Response, req *http.Request) Resu
...
-	case bool(klog.V(10)):
+	case bool(klog.V(5)):
 		return body
-	case bool(klog.V(9)):
+	case bool(klog.V(4)):
 		max = 10240
-	case bool(klog.V(8)):
+	case bool(klog.V(3)):
 		max = 1024
 	}
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/rest/request.go]: /hold

on this change... trace-level request logging is intentionally at much higher levels. mixing request-level trace logs with logic-level trace logs ("did x", "did y", etc) would make the logic-level trace logs almost impossible to consume

File: staging/src/k8s.io/client-go/rest/request.go [hunk 2]
```diff
@@ -1026,7 +1026,7 @@ func truncateBody(body string) string {
 // allocating a new string for the body output unless necessary. Uses a simple heuristic to determine
 // whether the body is printable.
 func glogBody(prefix string, body []byte) {
-	if klog.V(8) {
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/rest/request.go]: same here... logging all request bodies is far more verbose than other V(3) usage

File: staging/src/k8s.io/client-go/tools/clientcmd/loader.go [hunk 3]
```diff
@@ -372,7 +372,7 @@ func LoadFromFile(filename string) (*clientcmdapi.Config, error) {
 	if err != nil {
 		return nil, err
 	}
-	klog.V(6).Infoln("Config loaded from file: ", filename)
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/tools/clientcmd/loader.go]: this is far more frequent and far less important than other V(3) usage

File: staging/src/k8s.io/client-go/transport/round_trippers.go [hunk 4]
```diff
@@ -67,13 +67,13 @@ func HTTPWrappersForConfig(config *Config, rt http.RoundTripper) (http.RoundTrip
...
-	case bool(klog.V(8)):
+	case bool(klog.V(4)):
 		rt = newDebuggingRoundTripper(rt, debugJustURL, debugRequestHeaders, debugResponseStatus, debugResponseHeaders)
-	case bool(klog.V(7)):
+	case bool(klog.V(3)):
 		rt = newDebuggingRoundTripper(rt, debugJustURL, debugRequestHeaders, debugResponseStatus)
-	case bool(klog.V(6)):
+	case bool(klog.V(2)):
 		rt = newDebuggingRoundTripper(rt, debugURLTiming)
 	}
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/transport/round_trippers.go]: /hold

on this change... trace-level request logging is intentionally at much higher levels. mixing request-level trace logs with logic-level trace logs ("did x", "did y", etc) would make the logic-level trace logs almost impossible to consume

**Comment by liggitt** [comment]
Thanks for the PR, it's not clear that the described standard will work properly with the volume and size of the request tracing logs. Let's take the discussion back to the parent issue before proceeding here.

File: staging/src/k8s.io/client-go/transport/round_trippers.go (hunk 4 above)
Comment (RyanRod) [reply staging/src/k8s.io/client-go/transport/round_trippers.go]: For a switch on the log level such as this, is it then better to flatten it out to only be 4/5 verbosity (removing some of the cases), or just keep it outside of the typical 1-5 verbosity levels like it already is?

Comment (liggitt) [reply staging/src/k8s.io/client-go/transport/round_trippers.go]: Each level is useful for getting increasingly more trace information (url, headers, truncated body, full body). I don't see the benefit in flattening or removing cases.

**Comment by fedebongio** [comment]
/assign @liggitt 
(thanks for looking into it)

**Pull request closed without merging**